modules/programs/vim-vint.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.vim-vint;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+in {
+  meta.maintainers = [ maintainers.tomodachi94 ];
+
+  options = {
+    programs.vim-vint = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        defaultText = "false";
+        description = ''
+          Whether to enable the Vint linter for Vimscript.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.vim-vint;
+        defaultText = literalExpression "pkgs.vim-vint";
+        example =
+          literalExpression "(pkgs.vim-vint.override { enableCheck = true; })";
+        description = ''
+          The <literal>vim-vint</literal> package to use.
+        '';
+      };
+
+      settings = mkOption {
+        type = yamlFormat.type;
+        default = { };
+        description = ''
+          Configuration written to
+          <filename>$HOME/.vintrc.yaml</filename>
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    xdg.configFile.".vintrc.yaml".source = # Vint's configuration location isn't very smart. It checks $XDG_CONFIG_HOME for `.vintrc.yaml`, then it checks $HOME.
```

Comment:
The comment should go on its own line (or two lines). I'd just leave it out though.